extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeMixPlaylistExtractor.java

```diff
@@ -0,0 +1,173 @@
+import org.schabi.newpipe.extractor.exceptions.ParsingException;
+import org.schabi.newpipe.extractor.linkhandler.ListLinkHandler;
+import org.schabi.newpipe.extractor.localization.TimeAgoParser;
+import org.schabi.newpipe.extractor.playlist.PlaylistExtractor;
+import org.schabi.newpipe.extractor.stream.StreamInfoItem;
+import org.schabi.newpipe.extractor.stream.StreamInfoItemsCollector;
+
+/**
+ * A YoutubePlaylistExtractor for a mix (auto-generated playlist). It handles urls in the format of
+ * "youtube.com/watch?v=videoId&list=playlistId"
+ */
+public class YoutubeMixPlaylistExtractor extends PlaylistExtractor {
+
+
+    private final static String CONTENTS = "contents";
+    private final static String RESPONSE = "response";
+    private final static String PLAYLIST = "playlist";
+    private final static String TWO_COLUMN_WATCH_NEXT_RESULTS = "twoColumnWatchNextResults";
+    private final static String PLAYLIST_PANEL_VIDEO_RENDERER = "playlistPanelVideoRenderer";
+
+    private JsonObject playlistData;
+
+    public YoutubeMixPlaylistExtractor(StreamingService service, ListLinkHandler linkHandler) {
+        super(service, linkHandler);
+    }
+
+    @Override
+    public void onFetchPage(@Nonnull Downloader downloader)
+        throws IOException, ExtractionException {
+        final String url = getUrl() + "&pbj=1";
+        final JsonArray ajaxJson = getJsonResponse(url, getExtractorLocalization());
+        JsonObject initialData = ajaxJson.getObject(3).getObject(RESPONSE);
+        playlistData = initialData.getObject(CONTENTS).getObject(TWO_COLUMN_WATCH_NEXT_RESULTS)
+            .getObject(PLAYLIST).getObject(PLAYLIST);
+    }
+
+    @Nonnull
+    @Override
+    public String getName() throws ParsingException {
+        final String name = playlistData.getString("title");
+        if (name == null) {
+            throw new ParsingException("Could not get playlist name");
+        }
+        return name;
+    }
+
+    @Override
+    public String getThumbnailUrl() throws ParsingException {
+        try {
+            final String playlistId = playlistData.getString("playlistId");
+            return getThumbnailUrlFromId(playlistId);
+        } catch (Exception e) {
+            throw new ParsingException("Could not get playlist thumbnail", e);
+        }
+    }
+
+    @Override
+    public String getBannerUrl() {
+        return "";
+    }
+
+    @Override
+    public String getUploaderUrl() {
+        //Youtube mix are auto-generated
+        return "";
+    }
+
+    @Override
+    public String getUploaderName() {
+        //Youtube mix are auto-generated by YouTube
+        return "YouTube";
+    }
+
+    @Override
+    public String getUploaderAvatarUrl() {
+        //Youtube mix are auto-generated by YouTube
+        return "https://upload.wikimedia.org/wikipedia/commons/thumb/0/09/YouTube_full-color_icon_(2017).svg/320px-YouTube_full-color_icon_(2017).svg.png";
```

Comment:
When using YouTube's logo here, we can also use in other places. We should check legal stuff first. CC @TheAssassin

Reply:
I think this should be handled differently in the frontend by not displaying an avatar. @Stypox, @wb9688 What do you think?

Reply:
IANAL. However, using any brand icons is a bad idea, and might get us into trouble. I don't think it's worth the effort. I haven't had a look whether the linked file would be fine copyright wise either.

Reply:
Ok, so either we put no avatar (which would probably look strange) or we choose an icon from https://material.io/icons that somewhat represents the nature of the playlist and the uploader. I thought of these icons:
- [Radio](https://material.io/resources/icons/?icon=radio): <img src="https://raw.githubusercontent.com/google/material-design-icons/master/av/2x_web/ic_radio_white_48dp.png" width="20"/>
- [Auto renew](https://material.io/resources/icons/?icon=autorenew): <img src="https://raw.githubusercontent.com/google/material-design-icons/master/action/2x_web/ic_autorenew_white_48dp.png" href="" width="20"/>

Reply:
@XiangRongLin what do you think about this?

Reply:
I'm for either the Radio icon or no icon. Radio since the mix is basically a radio.
But that depends on which looks less out of place.

Reply:
whatever icons we use, that should be set in the app. It might be okay to return the URL to YouTube's logo in the extractor. However, it should be replaced in the frontend. Our extractor is used by other applications, too. They might want to use the original logo.

Reply:
Ok, that seems a good idea

Reply:
I don't think we should return a link to YouTube's logo.

Reply:
My vote goes to the Radio icon as well.

Reply:
Then i would be for returning a constant and each frontend can decide themself what they want to display based on the constant and legal stuff.
For NewPipe it would then be the radio icon.

Reply:
We should not use the wikimedia image though. Here are the icons which are used on the website when the browser does not support SVG:
https://s.ytimg.com/yts/img/favicon_144-vfliLAfaB.png
https://s.ytimg.com/yts/img/favicon_96-vflW9Ec0w.png
https://s.ytimg.com/yts/img/favicon_48-vflVjB_Qk.png
https://s.ytimg.com/yts/img/favicon_32-vflOogEID.png

> Then i would be for returning a constant and each frontend can decide themself what they want to display based on the constant and legal stuff.

@XiangRongLin What constant? Can you elaborate the idea? One can check whether the retrieved PlaylistInfo is a YTMix by using `YouTubeParsingHelper.isYouTubeMixId(id)`

Reply:
I meant as an arbitary string like `public static final String YOUTUBE_LOGO = "youtube_logo"`

Nice finding the png, i somehow only found svg.

Reply:
@gzsombor: We would like to know your opinion on this as well.

I personally think it would be best to return "" here, but @TobiGr thinks we should return a link to YouTube's logo, as we always return a URL for all other kind of playlists, however @TheAssassin thinks we should avoid logos/brands.

Either way, in the NewPipe front-end, we're gonna check if service ID equals YouTube's service ID and call `YoutubeParsingHelper`'s function to check whether the playlist ID is a mix. When that's the case, NewPipe will replace it with a radio icon embedded in our app.

Reply:
Why even return anything here?

This should return an empty string just like the uploader url and name, and [this is actually already handled by the front end](https://github.com/TeamNewPipe/NewPipe/blob/0a412e2abbc1d4cd50439ec89070a69a794af7ba/app/src/main/java/org/schabi/newpipe/fragments/list/playlist/PlaylistFragment.java#L265-L281).

Reply:
@gzsombor As a user of the extractor, what solution do you prefer?
The YouTube logo, an empty string or a constant?

@mauriciocolli @wb9688 I'd be okay with returning nothing, too

Reply:
What about returning a AvatarIcon class, something like this:
```
class AvatarIcon {
   String url;
   IconType type;
   enum IconType { YOUTUBE_LOGO, OTHER_LOGO, .... }
}

```

Either the url or the type would be filled - if the URL the frontend could use that, if not, it should revert to some custom magic.

Reply:
@gzsombor That's basically done in https://github.com/TeamNewPipe/NewPipeExtractor/pull/268. So we should wait for this PR to be merged. Let's go with the empty string for now and change this later in or after #268


@XiangRongLin Can you change this to 
```suggestion
        return "";
```

Reply:
Ok, I was about to ask ;-)